servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -155,12 +158,29 @@ public void deleteBranch(String branchName, String hash) throws NessieConflictEx
   }
 
   @Override
-  public LogResponse getCommitLog(String ref, Integer maxEntriesHint, String pageToken) throws NessieNotFoundException {
+  public LogResponse getCommitLog(String ref, Integer maxEntriesHint, String pageToken, @Nullable String author, @Nullable String committer,
+      @Nullable String after, @Nullable String before) throws NessieNotFoundException {
     int max = Math.min(maxEntriesHint != null ? maxEntriesHint : MAX_COMMIT_LOG_ENTRIES, MAX_COMMIT_LOG_ENTRIES);
     Hash startRef = getHashOrThrow(pageToken != null ? pageToken : ref);
+    final Instant afterInstant = parseToInstant(after);
+    final Instant beforeInstant = parseToInstant(before);
+
     try (Stream<ImmutableCommitMeta> s = getStore().getCommits(startRef).limit(max + 1)
         .map(cwh -> cwh.getValue().toBuilder().hash(cwh.getHash().asString()).build())) {
-      List<CommitMeta> items = s.collect(Collectors.toList());
+      Stream<ImmutableCommitMeta> commits = s;
```

Comment:
I think, the algorithm here is wrong now. For example, if `max==10` and you filter out one or more commits below, the `if (items.size() == max + 1) {` will *not* trigger and return the current page as the *last* page so the client will not ask for more pages.

The filtering probably requires the removal of the `.limit(max + 1)` above and a slightly changed logic below.

Reply:
It might be more obvious if filtering is extracted out in some separate method/class and the logic can be truly unit tested?

Reply:
you're totally right :man_facepalming:

Reply:
should be fixed now